### DISCUSSION TITLE
Updating  `mach` when using MLJ interface

### DISCUSSION THREAD

**Discussion by tomaklutfu** [MilesCranmer/PySR#760] [Q&A] [answered]
I use `MLJ` interface like this
```julia
structure = TemplateStructure{(:f, :g, :ω)}(
  ((; f, g, ω), (t, ω₀)) -> begin
    Ω = ω(ω₀)
    return f(ω₀, Ω*t) + g(ω₀, Ω)
  end
)
model = SRRegressor(;
    binary_operators       = (+, -, *, /),
    unary_operators        = (sin, cos),
    niterations            = 30,
    maxsize                = 30,
    batching               = true,
    batch_size             = 256,
    parallelism            = :multithreading,
    expression_type        = TemplateExpression,
    expression_options     = (; structure),
)

mach = machine(model, X, y)
fit!(mach)
```
I was able to update `mach` by `fit!(mach)` in a loop (mutating `X` and `y`) before v1.0(was not using `TemplateExpression` then). However, now, it just skips doing anything in `fit!` if used once and needs `MLJ.update` defined for updating `mach`. Is there a way to update `mach` or is there a way to search equations iteratively using different samples for `X`, and `y`?

**Comment by MilesCranmer** [comment, marked as answer]
So apparently this is the “correct” behavior for MLJ, because no hyperparams were updated between fit calls. The behavior before was apparently incorrect. (This behavior was implemented as a breaking change in 1.0.0 of SR.jl)

The way to do it now is simply increase the `niterations` and then call `fit!` again. It will fit for the _remaining_ iterations, given the previous search. So:

```julia
fit!(mach)
mach.model.niterations += 100
fit!(mach)
```

**Reply by tomaklutfu** [reply]
Thanks. That works.
